x-pack/plugins/ml/public/application/explorer/alerts_panel.tsx

```diff
@@ -0,0 +1,152 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiBasicTableColumn,
+  EuiInMemoryTable,
+  EuiNotificationBadge,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
+import useObservable from 'react-use/lib/useObservable';
+import { Axis, Chart, LineSeries, Position, ScaleType, Settings } from '@elastic/charts';
+import { CollapsiblePanel } from '../components/collapsible_panel';
+import { AnomalyDetectionAlert } from './anomaly_detection_alerts_state_service';
+import { useFieldFormatter } from '../contexts/kibana';
+import { useAnomalyExplorerContext } from './anomaly_explorer_context';
+import { Y_AXIS_LABEL_WIDTH } from './swimlane_annotation_container';
+
+export const AlertsPanel: FC = () => {
+  const [isOpen, setIsOpen] = useState(true);
+  const dateFormatter = useFieldFormatter(FIELD_FORMAT_IDS.DATE);
+  const durationFormatter = useFieldFormatter(FIELD_FORMAT_IDS.DURATION);
+
+  const { anomalyTimelineStateService, anomalyDetectionAlertsStateService } =
+    useAnomalyExplorerContext();
+
+  const alertsData = useObservable(anomalyDetectionAlertsStateService.anomalyDetectionAlerts$);
+  const selectedAlertsData = useObservable(anomalyDetectionAlertsStateService.selectedAlerts$);
+  const countByStatus = useObservable(anomalyDetectionAlertsStateService.countByStatus$);
+
+  const annotationXDomain = useObservable(anomalyTimelineStateService.timeDomain$);
+
+  const swimlaneContainerWidth = useObservable(
+    anomalyTimelineStateService.getContainerWidth$(),
+    anomalyTimelineStateService.getContainerWidth()
+  );
+
+  const columns: Array<EuiBasicTableColumn<AnomalyDetectionAlert>> = [
+    {
+      field: 'ruleName',
+      name: 'Rule name',
+      sortable: true,
+      truncateText: false,
+    },
+    {
+      field: 'jobId',
+      name: 'Job ID',
+      sortable: true,
+      truncateText: false,
+    },
+    {
+      field: 'anomalyTimestamp',
+      name: 'Anomaly time',
+      sortable: true,
+      truncateText: false,
+      render: (value: number) => dateFormatter(value),
+    },
+    {
+      field: 'timestamp',
+      name: 'Triggered at',
+      sortable: true,
+      truncateText: false,
+      render: (value: number) => dateFormatter(value),
+    },
+    {
+      name: 'Duration',
+      truncateText: false,
+      render: (value: AnomalyDetectionAlert) =>
+        durationFormatter(value.end_timestamp - value.timestamp),
+    },
+  ];
+
+  const yValueMap: Record<string, number> = {};
+  const chartData = alertsData
+    ?.map((alert, i) => {
+      if (!yValueMap.hasOwnProperty(alert.ruleName)) {
+        yValueMap[alert.ruleName] = Math.max(...Object.values(yValueMap), 0) + 1;
```

Comment:
Can we add a comment to provide additional context for the +1 here?

Reply:
This chart has been removed in favour of a line date histogram powered by the Lens embeddable.